src/system.rs

```diff
@@ -360,6 +398,13 @@ impl TypeStore {
             .or_else(|| self.imported_type_ids_by_name.get(name))
     }
 
+    /// Provides the [TypeId] associated with given [IonType] if it exists in the [TypeStore]  
+    /// Otherwise returns None
+    pub fn get_core_type_id_by_ion_type(&self, ion_type: &IonType) -> Option<&TypeId> {
+        self.core_type_ids_by_ion_type
+            .get(&format!("{:?}", ion_type))
```

Comment:
It would also be nice to avoid allocating/formatting the text each time we want to look up a core type; they're likely to be commonly referenced. Storing the `IonType` instead of a `String` will address this.

Reply:
Reference issue: #50